File: AppStore_InfoApp.class.m
// Developer by : Azozz ALFiras

@interface AppStoreInfoApp 
- (void)sentRequestWithBundleId:(NSString *)bundleId completion:(void (^)(NSDictionary * _Nullable, NSError * _Nullable))completion;
- (void)getInfoAppWithBundleId:(NSString *)bundleId version:(NSString *)version completion:(void (^)(NSDictionary * _Nullable))completion;
- (NSDictionary *)responseApiWithStatus:(NSString *)status url:(NSString *)url;
@end 

@implementation AppStoreInfoApp

- (void)sentRequestWithBundleId:(NSString *)bundleId completion:(void (^)(NSDictionary * _Nullable, NSError * _Nullable))completion {
    NSString *urlString = [NSString stringWithFormat:@"http://itunes.apple.com/lookup?bundleId=%@", bundleId];
    NSURL *url = [NSURL URLWithString:urlString];
    
    NSURLSessionDataTask *task = [[NSURLSession sharedSession] dataTaskWithURL:url completionHandler:^(NSData * _Nullable data, NSURLResponse * _Nullable response, NSError * _Nullable error) {
        if (error) {
            completion(nil, error);
            return;
        }
        
        NSError *jsonError;
        NSDictionary *json = [NSJSONSerialization JSONObjectWithData:data options:0 error:&jsonError];
        
        if (!jsonError) {
            completion(json, nil);
        }
    }];
    
    [task resume];
}

- (void)getInfoAppWithBundleId:(NSString *)bundleId version:(NSString *)version completion:(void (^)(NSDictionary * _Nullable))completion {
    [self sentRequestWithBundleId:bundleId completion:^(NSDictionary * _Nullable data, NSError * _Nullable error) {
        if (!error) {
            NSString *vAppStore = data[@"version"];
            NSString *appLink = data[@"trackViewUrl"];
            
            if ([vAppStore isEqualToString:version]) {
                completion([self responseApiWithStatus:@"Yes" url:appLink]);
            } else {
                completion([self responseApiWithStatus:@"No" url:appLink]);
            }
        }
    }];
}

- (NSDictionary *)responseApiWithStatus:(NSString *)status url:(NSString *)url {
    if ([status isEqualToString:@"Yes"]) {
        return @{
            @"status": @"success",
            @"status_message": @"There is no application update"
        };
    } else {
        return @{
            @"status": @"failed",
            @"status_message": @"The version does not match. An update is required",
            @"app_link": url
        };
    }
}

@end

File: AppStore_InfoApp.class.swift
// Developer by : Azozz ALFiras

import Foundation

class AppStoreInfoApp {
    func sentRequest(bundleId: String, completion: @escaping (Result<[String: Any], Error>) -> Void) {
        let urlString = "http://itunes.apple.com/lookup?bundleId=\(bundleId)"
        guard let url = URL(string: urlString) else {
            return
        }
        
        URLSession.shared.dataTask(with: url) { (data, response, error) in
            if let error = error {
                completion(.failure(error))
                return
            }
            
            if let data = data,
               let json = try? JSONSerialization.jsonObject(with: data, options: []) as? [String: Any] {
                completion(.success(json))
            }
        }.resume()
    }

    func getInfoApp(bundleId: String, version: String, completion: @escaping (Result<[String: Any], Error>) -> Void) {
        sentRequest(bundleId: bundleId) { result in
            switch result {
            case .success(let data):
                if let vAppStore = data["version"] as? String, let appLink = data["trackViewUrl"] as? String {
                    if vAppStore == version {
                        completion(responseApi(status: "Yes", url: appLink))
                    } else {
                        completion(responseApi(status: "No", url: appLink))
                    }
                }
            case .failure(let error):
                completion(.failure(error))
            }
        }
    }

    func responseApi(status: String, url: String) -> [String: Any] {
        if status == "Yes" {
            return [
                "status": "success",
                "status_message": "There is no application update"
            ]
        } else {
            return [
                "status": "failed",
                "status_message": "The version does not match. An update is required",
                "app_link": url
            ]
        }
    }
}
